projectile.el

```diff
@@ -3342,13 +3365,15 @@ test file."
            "No matching source file found for project type `%s'"
            (projectile-project-type))))
     ;; find the matching test file
-    (let ((test-file (projectile-find-matching-test file-name)))
-      (if test-file
-          (projectile-expand-root test-file)
-        (if projectile-create-missing-test-files
-            (projectile-create-test-file-for file-name)
-          (error "No matching test file found for project type `%s'"
-                 (projectile-project-type)))))))
+    (let* ((test-file (projectile-find-matching-test file-name))
+           (test-file-or-fallback (or test-file (projectile--test-file-fallback file-name)))
```

Comment:
`projectile--test-file-fallback` here does the same as `projectile-create-test-file-for` did previously, minus the creation of the test file which has been moved to this function.

I'm undecided whether moving `projectile--test-file-fallback` to `projectile-find-matching-test` (as a final `when-let` :smile:) or keeping it here would make more sense. Let me know what you think.